src/atoms/Text/Text.web.js

```diff
@@ -1,3 +1,139 @@
-const Text = () => null;
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import baseTheme from '../../tokens/theme';
+import isDefined from '../../_helpers/isDefined';
+import { getColorKeys, getColor, getLineHeight } from '../../_helpers/theme';
+
+/*
+ * Sets height for the div
+ * @param {Object} theme
+ * @param {String} size
+ * @param {number} maxLines
+ */
+const calculateHeight = ({ theme, size, maxLines }) => {
+  const lineHeight = getLineHeight(theme, size);
+  const containerHeight = `${parseFloat(lineHeight) * maxLines}px`;
+  return containerHeight;
+};
+
+const styles = {
+  fontFamily({ theme, _weight }) {
+    return theme.fonts.family.lato[_weight];
+  },
+  fontSize({ theme, size }) {
+    return theme.fonts.size[size];
+  },
+  color({ theme, color }) {
+    return getColor(theme, color);
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+  letterSpacing({ theme, _letterSpacing }) {
+    return `${theme.fonts.letterSpacing[_letterSpacing]}`;
+  },
+  lineHeight({ theme, size, _lineHeight }) {
+    if (_lineHeight) {
+      return theme.fonts.lineHeight[_lineHeight];
+    } else {
+      return getLineHeight(theme, size);
+    }
+  },
+  maxHeight({ theme, _lineHeight, maxLines }) {
+    if (isDefined(maxLines)) {
+      const maxHeight = calculateHeight({ theme, _lineHeight, maxLines });
+      return maxHeight;
+    } else {
+      return 'initial';
+    }
+  },
+  whiteSpace({ maxLines }) {
+    if (isDefined(maxLines)) {
+      return 'initial';
+    } else {
+      return 'nowrap';
+    }
+  },
+  overflow({ maxLines }) {
+    if (isDefined(maxLines)) {
+      return 'hidden';
+    } else {
+      return 'initial';
+    }
+  },
+  textOverflow({ maxLines }) {
+    if (isDefined(maxLines)) {
+      return 'ellipsis';
+    } else {
+      return 'initial';
+    }
+  },
+  textDecoration({ _isUnderlined }) {
+    if (_isUnderlined) {
+      return 'undefined';
+    } else {
+      return 'none';
+    }
+  },
+  ellipses({ maxLines }) {
+    if (isDefined(maxLines)) {
+      return "'...'";
+    } else {
+      return '';
+    }
+  },
+};
+
+const Text = styled.div`
+  font-family: ${styles.fontFamily};
+  font-size: ${styles.fontSize};
+  color: ${styles.color};
+  text-decoration-line: ${styles.textDecoration};
```

Comment:
why `text-decoration-line` and not `text-decoration`? `text-decoration-line` is not supported on IE

Reply:
done